packages/@aws-cdk/aws-sns/lib/subscription-filter-policy.ts

```diff
@@ -0,0 +1,167 @@
+/**
+ * A SNS subscription filter policy.
+ */
+export class SubscriptionFilterPolicy {
```

Comment:
Feels like this can be a simple interface (the attributes map) and move this logic to the Subscription class.